mlflow/types/utils.py

```diff
@@ -480,7 +480,7 @@ def _infer_schema_from_type_hint(type_hint, examples=None):
 
 
 def _infer_type_and_shape(value):
-    if isinstance(value, (list, np.ndarray, pd.Series)):
```

Comment:
Curious why we added `pd.Series`?

Reply:
I don't quite remember 😂 But looks like I was trying to add all `list like` types, while this is not consistent and no use cases.

Reply:
What would happen if we support `pd.Series`?

Reply:
We need to infer the inner object type for pd.Series, and then cast it to built-in list or numpy array, I feel like this process is error-prone, and we should avoid type casting for customers in this situation unless there're common use cases.

Reply:
Got it. Let's remove it.